pipeline/matching/generators/limited_schedules.py

```diff
@@ -7,13 +7,78 @@
 
 
 class LimitedSchedulesGenerator(MatchGenerator):
-    def __init__(self, max_available: int = 0):
+    def __init__(self, max_available: int = 1):
         # Maximum number of available slots in a user's schedule for them to be
         # eligible for a limited availability match.
         self.max_available = max_available
         super().__init__(name=GENERATOR_LIMITED_SCHEDULES)
 
     def generate(self, inp: MatchingInput) -> Iterator[Match]:
         # TODO: Implement your generator
-        inp.logger.info("Luke Arrington was here!")
+
+        # get limited availability users
+        limited_availability = []
+        for user in inp.users:  # for users in the list users
+            if (
+                len(user.schedule) <= self.max_available
+            ):  # if the number of the user schedule is less than or equal to the max available which is one
+                limited_availability.append(
+                    user
+                )  # then consider that a user a limited schedule user and add to the list
+        inp.logger.info(limited_availability)
+
+        matches = {}
+        # compare limited users to all users
+        for limited_user in limited_availability:
+            for user in inp.users:
+                if (
+                    limited_user.uid != user.uid
+                ):  # check that we are not looking at the current limited_user
+                    # compare limited user schedule to schedule of user in all users
+                    for l_availability in limited_user.schedule:
+                        for availability in user.schedule:
+                            if (
+                                l_availability == availability
+                            ):  # if the availabilities in limited user schedule and other user schedule are the same, then we have a match!
```

Comment:
Right now, I think what you are doing is getting just one time the two users have in common. We want to get all the times they have in common. It should be a small list since one user has limited availability. But, since the parameter `max_available` can change, we shouldn't assume that it will always be `1`.

I would recommend creating a dictionary like `common_availability` where the key is the set of the two user IDs and the value is a list of the availabilities they have in common. When you go through this loop, you can append the matching availabilities to that list.